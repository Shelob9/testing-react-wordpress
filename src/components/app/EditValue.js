import React from 'react';
/**
 * Edit the value
 */
export const EditValue = ({ onChange, value,id,className}) => {
  const changeHandler = event => onChange(event.target.value);
  return (
      <div className={className}>
      <label htmlFor={id}>Set Value</label>
        <input data-testid={id} id={id} onChange={changeHandler} value={value} />
    </div>
  );
};