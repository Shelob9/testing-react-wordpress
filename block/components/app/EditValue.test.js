//Import React
import React from "react";
//Import test renderer
import { render, fireEvent, cleanup } from "@testing-library/react";
//Import component to test
import { EditValue } from "./EditValue";

describe("EditValue componet", () => {
  afterEach(cleanup);

  it("matches snapshot", () => {
    const { container } = render(
      <EditValue
        onChange={jest.fn()}
        value={"Hi Roy"}
        id={"some-id"}
        className={"some-class"}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("Calls the onChange function", () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <EditValue
        onChange={onChange}
        value={""}
        id={"input-test"}
        className={"some-class"}
      />
    );
    fireEvent.change(getByTestId("input-test"), {
      target: { value: "New Value" }
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it("Passes updated value, not event to onChange callback", () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <EditValue
        onChange={onChange}
        value={""}
        id="input-test"
        className={"some-class"}
      />
    );
    fireEvent.change(getByTestId("input-test"), {
      target: { value: "New Value" }
    });
    expect(onChange).toHaveBeenCalledWith("New Value");
  });
});
