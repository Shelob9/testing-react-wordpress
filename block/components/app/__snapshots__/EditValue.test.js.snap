// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditValue componet matches snapshot 1`] = `
<div>
  <div
    class="some-class"
  >
    <label
      for="some-id"
    >
      Set Value
    </label>
    <input
      data-testid="some-id"
      id="some-id"
      value="Hi Roy"
    />
  </div>
</div>
`;
